src/lib.rs

```diff
@@ -85,6 +85,15 @@
 //! It should work on any POSIX.1-2001 system, which are all the major big OSes with the notable
 //! exception of Windows.
 //!
+//! Non-standard signals are also supported. Pass the signal value directly from `libc` or use
+//! the numeric value directly.
+//! ```rust
+//!    use std::sync::Arc;
+//!    use std::sync::atomic::{AtomicBool};
+//!    let term = Arc::new(AtomicBool::new(false));
+//!    let _ = signal_hook::flag::register(libc::SIGINT, Arc::clone(&term));
+//! ```
```

Comment:
Just a small style nitpick:

* Do not indent inside the example. The rendered docs then contain the indentation too and it looks odd. There's no outer block to indent against.
* Please leave a newline between the paragraph and the example at the top.

Reply:
Excellent. I also figured out how to generate the documentation so now it looks like this:
<img width="1052" alt="image" src="https://user-images.githubusercontent.com/433817/72899801-998ed300-3cdb-11ea-95f2-5c2f4b3298e0.png">